azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -219,6 +219,7 @@ def save_to_history(etag, text):
                 pass
             except VmSettingsError as exception:
                 save_to_history(exception.etag, exception.vm_settings_text)
+                raise
```

Comment:
this error needs to be propagated to the caller, I missed that in a previous commit